django_app/redbox_app/redbox_core/admin.py

```diff
@@ -82,26 +82,35 @@ class CitationInline(admin.StackedInline):
     extra = 1
 
 
-class ChatMessageAdmin(admin.ModelAdmin):
-    list_display = ["text", "role", "get_user", "chat", "route", "created_at"]
+class ChatMessageAdmin(ExportMixin, admin.ModelAdmin):
+    list_display = ["short_text", "role", "get_user", "chat", "route", "created_at"]
     list_filter = ["role", "route", "chat__user"]
     date_hierarchy = "created_at"
     inlines = [CitationInline]
+    readonly_fields = ["selected_files", "source_files"]
 
     @admin.display(ordering="chat__user", description="User")
     def get_user(self, obj):
         return obj.chat.user
 
+    @admin.display(description="text")
+    def short_text(self, obj):
+        max_length = 128
+        if len(obj.text) < max_length:
+            return obj.text
+        return obj.text[: max_length - 3] + "..."
+
 
 class ChatMessageInline(admin.StackedInline):
     model = models.ChatMessage
     ordering = ("modified_at",)
-    readonly_fields = ["modified_at", "source_files"]
+    fields = ["text", "role", "route", "rating"]
+    readonly_fields = ["text", "role", "route", "rating"]
```

Comment:
as this is just an inline (and the ChatMessage has its own dedicated Admin) we dont need to see all details, just the crucial ones